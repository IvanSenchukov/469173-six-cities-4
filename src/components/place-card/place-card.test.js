import React from 'react';
import renderer from 'react-test-renderer';
import PlaceCard from './place-card.jsx';

const DEFAULT_NAME = `PlaceCard.name`;
const DEFAULT_TYPE = `PlaceCard.type`;
const DEFAULT_REFERENCE = `PlaceCard.reference`;
const DEFAULT_IMAGE_SRC = `PlaceCard.imageSrc`;
const DEFAULT_PRICE = 90;
const DEFAULT_PERIOD = `PlaceCard.period`;
const DEFAULT_RATING = 4;


it(`renders correctly with only required fields`, () => {
  const tree = renderer
    .create(<PlaceCard
      name={DEFAULT_NAME}
      type={DEFAULT_TYPE}
      reference={DEFAULT_REFERENCE}
      imageSrc={DEFAULT_IMAGE_SRC}
      price={DEFAULT_PRICE}
      period={DEFAULT_PERIOD}
      rating={DEFAULT_RATING}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`renders correctly with full rating`, () => {
  const tree = renderer
    .create(<PlaceCard
      name={DEFAULT_NAME}
      type={DEFAULT_TYPE}
      reference={DEFAULT_REFERENCE}
      imageSrc={DEFAULT_IMAGE_SRC}
      price={DEFAULT_PRICE}
      period={DEFAULT_PERIOD}
      rating={5}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`renders correctly when added to bookmarks`, () => {
  const tree = renderer
    .create(<PlaceCard
      name={DEFAULT_NAME}
      type={DEFAULT_TYPE}
      reference={DEFAULT_REFERENCE}
      imageSrc={DEFAULT_IMAGE_SRC}
      price={DEFAULT_PRICE}
      period={DEFAULT_PERIOD}
      rating={DEFAULT_RATING}
      addedToBookmarks={true}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`renders correctly when offer is premium`, () => {
  const tree = renderer
    .create(<PlaceCard
      name={DEFAULT_NAME}
      type={DEFAULT_TYPE}
      reference={DEFAULT_REFERENCE}
      imageSrc={DEFAULT_IMAGE_SRC}
      price={DEFAULT_PRICE}
      period={DEFAULT_PERIOD}
      rating={DEFAULT_RATING}
      premium={true}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
