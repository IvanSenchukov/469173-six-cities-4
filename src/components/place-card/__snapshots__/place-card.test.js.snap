// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when added to bookmarks 1`] = `
<article
  className="cities__place-card place-card"
>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="PlaceCard.reference"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="PlaceCard.imageSrc"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          90
        </b>
        <span
          className="place-card__price-text"
        >
          / 
          PlaceCard.period
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button--active button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="PlaceCard.reference"
      >
        PlaceCard.name
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      PlaceCard.type
    </p>
  </div>
</article>
`;

exports[`renders correctly when offer is premium 1`] = `
<article
  className="cities__place-card place-card"
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="PlaceCard.reference"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="PlaceCard.imageSrc"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          90
        </b>
        <span
          className="place-card__price-text"
        >
          / 
          PlaceCard.period
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="PlaceCard.reference"
      >
        PlaceCard.name
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      PlaceCard.type
    </p>
  </div>
</article>
`;

exports[`renders correctly with full rating 1`] = `
<article
  className="cities__place-card place-card"
>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="PlaceCard.reference"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="PlaceCard.imageSrc"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          90
        </b>
        <span
          className="place-card__price-text"
        >
          / 
          PlaceCard.period
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="PlaceCard.reference"
      >
        PlaceCard.name
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      PlaceCard.type
    </p>
  </div>
</article>
`;

exports[`renders correctly with only required fields 1`] = `
<article
  className="cities__place-card place-card"
>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="PlaceCard.reference"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="PlaceCard.imageSrc"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          90
        </b>
        <span
          className="place-card__price-text"
        >
          / 
          PlaceCard.period
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="PlaceCard.reference"
      >
        PlaceCard.name
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      PlaceCard.type
    </p>
  </div>
</article>
`;
